### PR TITLE
Upgrade to Clerk Core 2

### PR DESCRIPTION
**Pull request by wobsoriano**
Hi! Appreciate you doing this module.

This PR updates vue-clerk and h3-clerk versions, fix incorrect imports and updated the playground.

Update: This PR now upgrades the h3-clerk and vue-clerk to Clerk Core 2 versions

### CODE REVIEW

**Comment by RodrigoProjects** [comment]
Thanks for the updates! Will wait for the ready for review :)

File: src/runtime/plugin.ts [hunk 1]
```diff
@@ -7,6 +7,6 @@ export default defineNuxtPlugin(async (nuxtApp) => {
 
   nuxtApp.vueApp.use(clerkPlugin, {
     publishableKey,
-    options: NuxtClerkModule.vueClerkOptions,
+    ...NuxtClerkModule.vueClerkOptions,
```
Comment (wobsoriano) [inline src/runtime/plugin.ts]: `options` property is removed in favor of top-level options. See https://github.com/wobsoriano/vue-clerk/pull/26

File: src/module.ts [hunk 2]
```diff
@@ -26,11 +27,15 @@ export default defineNuxtModule<ModuleOptions>({
...
+    const { installServerMiddleware = true } = options
+
     addTemplate({
       filename: 'nuxt-clerk.mjs',
+      write: true,
       getContents: () => [
         'export default {',
-        ` vueClerkConfig: ${options.vueClerk ? JSON.stringify(options.vueClerk) : 'null'}`,
+        ` vueClerkOptions: ${options.vueClerk ? JSON.stringify(options.vueClerk) : 'null'},`,
+        ` h3ClerkOptions: ${options.vueClerk ? JSON.stringify(options.h3Clerk) : 'null'},`,
```
Comment (wobsoriano) [inline src/module.ts]: I assume this will not be included in client build?

File: src/module.ts [hunk 3]
```diff
@@ -50,20 +55,43 @@ export default defineNuxtModule<ModuleOptions>({
...
-      from: '@vue-clerk'
+      from: 'vue-clerk'
     })))
 
+    if (installServerMiddleware) {
+      addServerHandler({
+        middleware: true,
+        handler: resolve('./runtime/middleware')
+      })
+    }
```
Comment (wobsoriano) [inline src/module.ts]: This middleware is required for clerk to inject cookies and stuff. It's not recommended if SSR is set to false, though

File: src/module.ts [hunk 4]
```diff
@@ -26,11 +27,15 @@ export default defineNuxtModule<ModuleOptions>({
...
+    const { installServerMiddleware = true } = options
+
     addTemplate({
       filename: 'nuxt-clerk.mjs',
+      write: true,
       getContents: () => [
         'export default {',
-        ` vueClerkConfig: ${options.vueClerk ? JSON.stringify(options.vueClerk) : 'null'}`,
+        ` vueClerkOptions: ${options.vueClerk ? JSON.stringify(options.vueClerk) : 'null'},`,
+        ` h3ClerkOptions: ${options.h3Clerk ? JSON.stringify(options.h3Clerk) : 'null'},`,
```
Comment (wobsoriano) [inline src/module.ts:33]: I assume this will not be included in client bundle 👀

Comment (RodrigoProjects) [reply src/module.ts:33]: It all depends on where it is imported! Let me check

Comment (RodrigoProjects) [reply src/module.ts:33]: We should create a server and client plugin and only import the template of the config we need in each enviroment. 

Which means creating 2 templates, one for the client config and another for the server config if it makes sense

Comment (wobsoriano) [reply src/module.ts:33]: @RodrigoProjects would you be interested in doing that?

**Comment by wobsoriano** [comment]
@RodrigoProjects pushed more updates that will upgrade this to [Clerk Core 2](https://clerk.com/changelog/2024-04-19).

File: pnpm-lock.yaml
Comment (wobsoriano) [inline pnpm-lock.yaml:1]: We have a bun lockfile so I believe this is extra

File: package.json [hunk 5]
```diff
@@ -34,21 +34,21 @@
     "test:watch": "vitest watch"
   },
   "dependencies": {
-    "@nuxt/kit": "^3.9.0",
-    "@nuxtjs/tailwindcss": "^6.10.1",
-    "h3-clerk": "^0.3.8",
-    "vue-clerk": "^0.1.1"
+    "@nuxt/kit": "^3.12.1",
+    "@nuxtjs/tailwindcss": "^6.12.0",
```
Comment (wobsoriano) [inline package.json:38]: https://github.com/RodrigoProjects/nuxt-clerk/issues/4 - I agree with this. Do we really need tailwind as dep?

**Comment by wobsoriano** [comment]
Closing in favor of https://www.vue-clerk.com/guides/nuxt

**Pull request closed without merging**